app/modules/wifi.c

```diff
@@ -506,24 +475,59 @@ static int wifi_setip( lua_State* L, uint8_t mode )
   return 1;  
 }
 
-// Lua: realtype = sleeptype(type)
-static int wifi_sleeptype( lua_State* L )
+// Lua: wifi.sta.getaplist
+static int wifi_station_get_ap_info4lua( lua_State* L )
 {
-  unsigned type;
-  
-  if ( lua_isnumber(L, 1) ){
-    type = lua_tointeger(L, 1);
-    if ( type != NONE_SLEEP_T && type != LIGHT_SLEEP_T && type != MODEM_SLEEP_T )
-      return luaL_error( L, "wrong arg type" );
-    if(!wifi_set_sleep_type(type)){
-      lua_pushnil(L);
-      return 1;
-    }
+  char temp[64];
+  struct station_config config[5];
+  uint8 number_of_aps = wifi_station_get_ap_info(config);
+  lua_newtable(L);
+  lua_pushnumber(L, number_of_aps);
+  lua_setfield(L, -2, "qty");
+  WIFI_DBG("\n\t# of APs stored in flash:%d\n", number_of_aps);
+  WIFI_DBG("%-6s %-32s %-64s %-18s\n", "index:", "ssid:", "password:", "bssid:");
+  for(int i;i<number_of_aps;i++)
+  {
+    lua_newtable(L);
+    lua_pushstring(L, config[i].ssid);
+    lua_setfield(L, -2, "ssid");
+    lua_pushstring(L, config[i].password);
+    lua_setfield(L, -2, "password");
+    memset(temp, 0, sizeof(temp));
+    c_sprintf(temp, MACSTR, MAC2STR(config[i].bssid));
```

Comment:
I can't find either `c_snprintf()` or `vsnprintf()` in `app/libc/`.
I did find `ets_snprintf()` and `ets_vsnprintf()` in `libmain.a(eagle_lib.o)` via the mapfile, however  some of the format specifiers(fF eE gG aA) are not supported

Here is the output of the test:
```lua
	Signed integer(%d):-9001 (%i):-9001
	Unsigned integer(%u):9001
	Char(%c):'C'
	String(%s):"string"
	Hex(%x):dead (%X):BEEF
	Pointer(%p):4025965c
	Float(%f):%f
	Scientific notation(%e):%e (%E):%E
	Shortest representation(%g):%g (%G):%G
	Hexadecimal floating point(%a):%a (%A):%A
	Octal(%o):2329
```